### PR TITLE
Map types directive

### PR DESCRIPTION
**Pull request by eczajk1**
Proposed enhancement for maintaining order of many map-type layers.

[This fiddle](http://jsfiddle.net/eczajk1/kqa68ue1/) shows the new directive in action, compared to the fiddle shown in #1871.

I realize that this is a fairly heavy PR and people are busy, so thanks in advance for any feedback that anyone can offer.

### CODE REVIEW

**Comment by eczajk1** [comment]
Any advice on what it might take to get this merged?

**Comment by devotis** [comment]
I guess to resolve the merge conflicts. BTW, I think your fiddle that demonstrates how to add a Openstreetmap layer on top of a Google Map is great!

**Comment by nmccready** [comment]
@eczajk1 if you merge updates I'll merge this. I am not actively following the repo anymore. If you want I can give you privileges.

File: src/coffee/directives/api/models/parent/map-types-parent-model.coffee [hunk 1]
```diff
@@ -0,0 +1,24 @@
+angular.module('uiGmapgoogle-maps.directives.api.models.parent')
+.factory 'uiGmapMapTypesParentModel', [
+  'uiGmapBaseObject', 'uiGmapLogger', 'uiGmapMapTypeParentModel', 'uiGmapPropMap'
+  (BaseObject, Logger, MapTypeParentModel, PropMap) ->
+    class MapTypesParentModel extends BaseObject
+      constructor: (@scope, @element, @attrs, @gMap, @$log = Logger) ->
+        unless @attrs.mapTypes?
+          @$log.info('layers attribute for the map-types directive is mandatory. Map types creation aborted!!')
```
Comment (nmccready) [inline src/coffee/directives/api/models/parent/map-types-parent-model.coffee]: nice

**Comment by nmccready** [comment]
LGTM

**Comment by nmccready** [comment]
@eczajk1 pulled down locally and merged

**Pull request merged**